Mathlib/CategoryTheory/ChosenFiniteProducts.lean

```diff
@@ -0,0 +1,130 @@
+/-
+Copyright (c) 2024 Adam Topaz. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Adam Topaz
+-/
+import Mathlib.CategoryTheory.Monoidal.OfChosenFiniteProducts.Symmetric
+import Mathlib.CategoryTheory.Limits.Constructions.FiniteProductsOfBinaryProducts
+
+/-!
+# Categories with chosen finite products
+
+We introduce a class, `ChosenFiniteProducts`, which bundles explicit choices
+for a terminal object and binary products in a category `C`.
+This is primarily useful for categories which have finite products with good
+definitional properties, such as the category of types.
+
+Given a category with such an instance, we also provide the associated
+symmetric monoidal structure so that one can write `X ⊗ Y` for the explicit
+binary product and `𝟙_ C` for the explicit terminal object.
+
+# Projects
+
+- Construct an instance of chosen finite products in the category of affine scheme, using
+  the tensor product.
+- Construct chosen finite products in other categories appearing "in nature".
+
+-/
+
+namespace CategoryTheory
+
+universe v u
+
+/--
+An instance of `ChosenFiniteProducts C` bundles an explicit choice of a binary
+product of two objects of `C`, and a terminal object in `C`.
+
+Users should use the monoidal notation: `X ⊗ Y` for the product and `𝟙_ C` for
+the terminal object.
+-/
+class ChosenFiniteProducts (C : Type u) [Category.{v} C] where
+  /-- A choice of a limit binary fan for any two objects of the category.-/
+  product : (X Y : C) → Limits.LimitCone (Limits.pair X Y)
+  /-- A choice of a terminal object. -/
+  terminal : Limits.LimitCone (Functor.empty.{v} C)
+
+namespace ChosenFiniteProducts
+
+instance (C : Type u) [Category.{v} C] [ChosenFiniteProducts C] : MonoidalCategory C :=
```

Comment:
I'm not an expert on instances, but I believe that the instances `ChosenFiniteProducts C -> MonoidalCategory C`, `ChosenFiniteProducts C -> SymmetricCategory C` and `ChosenFiniteProducts C -> HasFiniteProducts C` should all get `(priority := 100)` since they always apply.